modules/aws/ssm.go

```diff
@@ -68,3 +73,127 @@ func NewSsmClientE(t testing.TestingT, region string) (*ssm.SSM, error) {
 
 	return ssm.New(sess), nil
 }
+
+// WaitForSsmInstanceE waits until the instance get registered to the SSM inventory.
+func WaitForSsmInstanceE(t testing.TestingT, awsRegion, instanceID string, timeout time.Duration) error {
+	timeBetweenRetries := 2 * time.Second
+	maxRetries := int(timeout.Seconds() / timeBetweenRetries.Seconds())
+	description := fmt.Sprintf("Waiting for %s to appear in the SSM inventory", instanceID)
+
+	key := aws.String("AWS:InstanceInformation.InstanceId")
+	filterType := aws.String("Equal")
+	values := aws.StringSlice([]string{instanceID})
+	_, err := retry.DoWithRetryE(t, description, maxRetries, timeBetweenRetries, func() (string, error) {
+		client := NewSsmClient(t, awsRegion)
+		req, resp := client.GetInventoryRequest(&ssm.GetInventoryInput{
+			Filters: []*ssm.InventoryFilter{
+				{
+					Key:    key,
+					Type:   filterType,
+					Values: values,
+				},
+			},
+		})
+
+		if err := req.Send(); err != nil {
```

Comment:
NIT: we typically call `client.GetInventory` as a one-liner, rather than creating the request and calling `Send` separately.

Reply:
I'm not sure I get what you mean here

Reply:
In your AWS SDK usage, you are doing all the API calls in two steps:

1. Create the request: e.g., `client.GetInventoryRequest(input)`.
1. Send the request: e.g., ,`req.Send()`

What I'm saying is that most of these API calls can be done in one line: e.g., `client.GetInventory(input)`.

Reply:
I had no idea this was possible with this version of the AWS SDK, thanks!